Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1269,4 +1288,18 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CFNotificationCenter
 }
 #endif
 
+#if PLATFORM(IOS_FAMILY)
+HardwareKeyboardState WebProcessPool::cachedHardwareKeyboardState()
+{
+    RELEASE_ASSERT(isMainRunLoop());
+    return m_hardwareKeyboardState;
+}
+
+void WebProcessPool::setCachedHardwareKeyboardState(HardwareKeyboardState hardwareKeyboardState)
+{
+    RELEASE_ASSERT(isMainRunLoop());
+    m_hardwareKeyboardState = hardwareKeyboardState;
```

Comment:
What updates this cached value if hardware keyboard availability changes?

Reply:
Ah, good catch, I will update this patch with an observer of this state.

Thanks for reviewing!